### PR TITLE
GREEN-42 add gasUsed and transaction root for returned block data

### PR DESCRIPTION
**Pull request by tonititi**
Add block gasUsed and transaction root to the returned block data

### CODE REVIEW

File: src/external/Collector.ts [hunk 1]
```diff
@@ -305,22 +320,30 @@ class Collector extends BaseExternal {
...
         })
         .catch((e) => {
           nestedCountersInstance.countEvent('collector', `getBlock-error tx ${e.message}`)
           console.error('collector.getBlock could not get txs for the block', e)
           return []
         })
+      // Now we have the block gas used then return it to the RPC method
+      resultBlock.gasUsed = blockGasUsed
+      // Start to calculate the transaction root fot this block
+      resultBlock.transactionsRoot = await this.calculateTransactionRoot(readableBlock.tranactions)
```
Comment (justin-shardeum) [inline src/external/Collector.ts]: readableBlock.tranactions <-- appears to have a typo here.  Should be "transactions"?

File: package.json [hunk 2]
```diff
@@ -36,6 +36,7 @@
     "got": "11.8.6",
     "jayson": "3.6.5",
     "jsonwebtoken": "9.0.0",
+    "merkle-patricia-tree": "^4.2.4",
```
Comment (mhanson-github) [inline package.json]: Remove carat in version.

Comment (justin-shardeum) [inline package.json]: Will this still work without the "^"?  I remember in our group code review, we said we'd prefer not to have it.

File: src/external/Collector.ts (hunk 1 above)
Comment (mhanson-github) [inline src/external/Collector.ts]: This appears to be misspelled.  Am not sure how this could have worked.

**Review by mhanson-github — DISMISSED** [review]
Remove ^ in version.

Address other issue with misspelled parameter.

Retest

**Comment by afostr** [comment]
~/shardus/npm-release/json-rpc-server$ npm run compile

> shardeum-json-rpc@1.0.0 compile
> tsc -p .

src/external/Collector.ts:20:27 - error TS2307: Cannot find module 'ethers' or its corresponding type declarations.

20 import { BigNumber } from 'ethers'
                             ~~~~~~~~


Found 1 error.

**Review by afostr — DISMISSED** [review]
build does not compile
~/shardus/npm-release/json-rpc-server$ npm run compile

> shardeum-json-rpc@1.0.0 compile
> tsc -p .

src/external/Collector.ts:20:27 - error TS2307: Cannot find module 'ethers' or its corresponding type declarations.

20 import { BigNumber } from 'ethers'
                             ~~~~~~~~


Found 1 error.

File: src/external/Collector.ts [hunk 3]
```diff
@@ -15,7 +15,9 @@ import { Err, NewErr, NewInternalErr } from './Err'
 import { nestedCountersInstance } from '../utils/nestedCounters'
 import { BlockCacheManager } from '../cache/BlockCacheManager'
 import { sleep } from '../utils'
-
+import { BaseTrie } from 'merkle-patricia-tree'
+import RLP from 'rlp'
+import { BigNumber } from 'ethers'
```
Comment (jairajdev) [inline src/external/Collector.ts]: Looks like the import lib has to be `@ethersproject/bignumber`

File: src/external/Collector.ts [hunk 4]
```diff
@@ -245,6 +247,19 @@ class Collector extends BaseExternal {
     return await this.inner_getBlock(blockSearchValue, blockSearchType, details)
   }
 
+  async calculateTransactionRoot(txns: String[]): Promise<String> {
```
Comment (jairajdev) [inline src/external/Collector.ts]: I think the correct type has to be `string`

**Comment by tonititi** [comment]
By changing the command npm ci to npm i in both smoke tests 10+10 and 25+5, these two tests were passed

25+5: https://devs-work.shardeum.org/job/Testing/job/Smoke/job/work_in_progress/job/wip_green_25_plus_5/2/
10+10: https://devs-work.shardeum.org/job/Testing/job/Smoke/job/work_in_progress/job/wip_test_green_team/2/

**Review by jairajdev — DISMISSED** [review]

**Review by jairajdev — APPROVED** [review]

**Pull request merged**